cmd/initialise.go

```diff
@@ -0,0 +1,189 @@
+// Copyright © 2019 Zappi
+
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package cmd
+
+import (
+	"bytes"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/kms"
+	"github.com/aws/aws-sdk-go/service/s3/s3manager"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"github.com/spf13/cobra"
+)
+
+var httpClient    http.Client
+
+// InitPayload holds a Vault init request.
+type InitPayload struct {
+	RecoveryShares    int `json:"recovery_shares"`
+	RecoveryThreshold int `json:"recovery_threshold"`
+	SecretShares      int `json:"secret_shares"`
+	SecretThreshold   int `json:"secret_threshold"`
+}
+
+// InitResponse holds a Vault init response.
+type InitResponse struct {
+	Keys               []string `json:"keys"`
+	KeysBase64         []string `json:"keys_base64"`
+	RootToken          string   `json:"root_token"`
+	RecoveryKeys       []string `json:"recovery_keys"`
+	RecoveryKeysBase64 []string `json:"recovery_keys_base64"`
+}
+
+// initialiseCmd represents the initialise command
+var initialiseCmd = &cobra.Command{
+	Use:   "initialise",
+	Short: "A brief description of your command",
+	Long: `A longer description that spans multiple lines and likely contains examples
+and usage of using your command. For example:
+
+Cobra is a CLI library for Go that empowers applications.
+This application is a tool to generate the needed files
+to quickly create a Cobra application.`,
+	Run: func(cmd *cobra.Command, args []string) {
+		// make vault init request and get root token
+		fmt.Println("Initialising Vault...")
+		rootToken := initVault()
+		fmt.Println("Initialisation complete")
+
+		// AWS setup
+		region := "us-east-1"
```

Comment:
This could be parametrised into a flag whose value defaults to `us-east-1`. You don't have to do this now. But file it as an issue that can be closed when done.

Reply:
Can do. I was wondering about whether to put [this config value](https://github.com/Intellection/vault-init/pull/1/files#diff-6bf4c7bc7dd75a561cdd357f83883607R124) in a flag as well instead of getting from the env vars. Thoughts?

Reply:
Ok with envars as well.

Reply:
Alright. Put this in an env var as well
https://github.com/Intellection/vault-init/pull/1/commits/c84af2aa15a2064942d569c4164832c2a78352ec